File: client/app/snow/Snowfall.js
export default function Snowfalls({ snowfalls = [] }) {
  return (
    <div className="snowfall">
      <h2>Snowfall</h2>
      <ul className="snowfalls">
        {snowfalls.map((s) => (
          <li key={s.since}>
            <span className="snowfall-depth">{s.depth}</span>{' '}
            <span className="snowfall-since">{s.since}</span>
          </li>
        ))}
      </ul>
    </div>
  )
}

File: client/src/components/snowfall.rs
use shared::Snowfall;
use yew::prelude::*;

#[derive(Properties, PartialEq)]
pub struct Props {
    pub snowfalls: Vec<Snowfall>,
}

#[function_component(SnowfallView)]
pub fn snowfall_view(props: &Props) -> Html {
    html! {
        <div class="snowfall">
            <h2>{"Snowfall"}</h2>
            <ul class="snowfalls">
                { for props.snowfalls.iter().map(|s| html! {
                    <li key={s.since.clone()}>
                        <span class="snowfall-depth">{format!("{}\"", s.depth)}</span>
                        {" "}
                        <span class="snowfall-since">{&s.since}</span>
                    </li>
                })}
            </ul>
        </div>
    }
}
